web/permissions/permissions.go

```diff
@@ -49,50 +74,143 @@ const ContextClaims = "token_claims"
 // Extractor extracts the permission set from the context
 func Extractor(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		instance := middlewares.GetInstance(c)
-		var claims permissions.Claims
-		var err error
-
-		if token := getBearerToken(c); token != "" {
-			err = crypto.ParseJWT(token, keyPicker(instance), &claims)
-		} else if token := getQueryToken(c); token != "" {
-			err = crypto.ParseJWT(token, keyPicker(instance), &claims)
-		} else {
-			// no token is provided, hopefully the next handler does not need one
-			return next(c)
-		}
 
-		if err != nil {
+		_, _, err := extract(c)
+		// if no token is provided, we call next anyway,
+		// hopefully the next handler does not need permissions
+		if err != nil && err != ErrNoToken {
 			return err
 		}
 
-		if claims.Issuer != instance.Domain {
-			// invalid issuer in token
-			return permissions.ErrInvalidToken
-		}
+		return next(c)
+	}
+}
+
+func extractJWTClaims(c echo.Context, instance *instance.Instance) (*permissions.Claims, error) {
+	var claims permissions.Claims
+	var err error
+	if token := getBearerToken(c); token != "" {
+		err = crypto.ParseJWT(token, keyPicker(instance), &claims)
+	} else if token := getQueryToken(c); token != "" {
+		err = crypto.ParseJWT(token, keyPicker(instance), &claims)
```

Comment:
This is not an issue raised by this PR, but the `ErrInvalidAudience` and `ErrInvalidToken`, and maybe some other errors that would be returned by `ParseJWT`, should be treated so that they do not cause an HTTP error `500` (probably a `401`...)